src/utils/Base64URL.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import "openzeppelin-contracts/contracts/utils/Base64.sol";
+
+library Base64URL {
+    function encode(bytes memory data) public pure returns (string memory) {
+        string memory strb64 = Base64.encode(data);
```

Comment:
alternative approach to consider:
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Base64.sol

^ base64 is pretty small. i think we could turn it into Base64URL by:

-  changing two characters in the table
- removing the lines about adding equals signs 
- computing a correct (no equals padding) length for the result string

...so ours would be a small diff on theirs and actually shorter

wonder if OZ would include it in a future release, would fit imo

Reply:
no strong opinion either way, but I slightly prefer this so we're not touching any assembly / hand wrangling of bytes ourselves. Will go with this for now unless you feel strongly.